src/utils/utils.ts

```diff
@@ -50,12 +50,20 @@ export const apiFetchCall =
 
 export const createDatesArray = (
   firstDate: Date,
-  secondDate: Date
+  secondDate: Date,
+  intervalMinutes: number = config.timeBetweenCyclesInMinutes
 ): String[] => {
-  const intervalMS = config.timeBetweenCyclesInMinutes * 60 * 1000
+  if (!(firstDate instanceof Date) || !(secondDate instanceof Date)) {
+    console.log('Date format error')
+    return []
+  }
+  const intervalMS = intervalMinutes * 60 * 1000
   const firstDateMS = firstDate.getTime()
   const secondDateMS = secondDate.getTime()
-
+  if (isNaN(firstDateMS) || isNaN(secondDateMS)) {
+    console.log('Date format error')
+    return []
+  }
```

Comment:
I don't think this check is necessary. We already check if it's a valid Date object on line 56.

Reply:
Unfortunately, using instanceof to check if it's a date will allow any form of date to pass.  So new Date('random string') will register as a Date object with the value of Invalid Date, so even though it isn't a real date, it passes the date check.  By checking if .getTime is successful in returning a number, it is possible to check if the dates are accurate dates or not.  There isn't a good built in way to type check dates.

Reply:
Oh interesting, thanks for clarifying! In that case, I would combine the if statements on line 63 and 67 together like so:
```suggestion
  if (isNaN(firstDateMS) || isNaN(secondDateMS || firstDateMS >= secondDateMS)) {
    console.log('Date format error')
    return []
  }
```